44/README.md

```diff
@@ -0,0 +1,140 @@
+1. To work on an issue, a Contributor SHALL fork the project repository and then work on their forked repository.
+1. To submit a patch, a Contributor SHALL create a Platform pull request back to the project.
+1. A Contributor SHALL NOT commit changes directly to the project.
+1. If the Platform implements pull requests as issues, a Contributor MAY directly send a pull request without logging a separate issue.
+1. To discuss a patch, people MAY comment on the Platform pull request, on the commit, or elsewhere.
+1. To accept or reject a patch, a Maintainer SHALL use the Platform interface.
+1. Maintainers SHOULD NOT merge their own patches except in exceptional cases, such as non-responsiveness from other Maintainers for an extended period (more than 1-2 days).
+1. Maintainers SHALL NOT make value judgments on correct patches.
+1. Maintainers SHALL merge correct patches from other Contributors rapidly.
+1. Maintainers MAY merge incorrect patches from other Contributors with the goals of (a) ending fruitless discussions, (b) capturing toxic patches in the historical record, (c) engaging with the Contributor on improving their patch quality.
+1. The user who created an issue SHOULD close the issue after checking the patch is successful.
+1. Any Contributor who has value judgments on a patch SHOULD express these via their own patches.
+1. Maintainers SHOULD close user issues that are left open without action for an uncomfortable period of time.
+
+### 2.5. Branches and Releases
+
+1. The project SHALL have one branch ("master") that always holds the latest in-progress version and SHOULD always build.
+1. The project SHALL NOT use topic branches for any reason. Personal forks MAY use topic branches.
+1. To make a stable release a Maintainer shall tag the repository. Stable releases SHALL always be released from the repository master.
+
+### 2.6. Evolution of Public Contracts
+
+1. All Public Contracts (APIs or protocols) SHALL be documented.
+1. All Public Contracts SHOULD have space for extensibility and experimentation.
+1. A patch that modifies a stable Public Contract SHOULD not break existing applications unless there is overriding consensus on the value of doing this.
+1. A patch that introduces new features SHOULD do so using new names (a new contract).
+1. New contracts SHOULD be marked as "draft" until they are stable and used by real users.
+1. Old contracts SHOULD be deprecated in a systematic fashion by marking them as "deprecated" and replacing them with new contracts as needed.
+1. When sufficient time has passed, old deprecated contracts SHOULD be removed.
+1. Old names SHALL NOT be reused by new contracts.
+1. A new contract marked as "draft" MUST not be changed to "stable" until all the following conditions are met:
+    1. Documentation has been written and finalised.
+    1. Self-tests exercising the functionality are passing on the principle target platform.
```

Comment:
What is the "principle target platform", e.g. of libzmq? Shouldn't this at least be plural: principle target platform*s*?

What does "self-test" mean, compared with "test"? Do you mean "automated test"? 

I would encourage, but maybe not require, that tests are executed on the CI environment.

Reply:
Here I just copied the existing wording from the above paragraph 2.3.6 to maintain consistency:

> A patch MUST compile cleanly and pass project self-tests on at least the principle target platform.

Self-test is how the tests are called in zproject-based programs, I think that's where the term comes from.

I think it was left intentionally open as it varies wildly between projects - so it should be defined in the project itself rather than in the general process.

But this topic of tests/CI is going to be revised soon as part of related work so I'll leave it for that if it's OK

Reply:
That's fine. I was not aware that these terms were already used.